src/esssans/isis/io.py

```diff
@@ -0,0 +1,76 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+"""
+File loading functions for ISIS data, NOT using Mantid.
+"""
+from typing import NewType, TypeVar
+
+import sciline
+import scipp as sc
+
+from ..types import RunType
+
+PixelMaskFilename = NewType('PixelMaskFilename', str)
+CalibrationFilename = NewType('CalibrationFilename', str)
+
+FilenameType = TypeVar('FilenameType', bound=str)
+
+
+DataFolder = NewType('DataFolder', str)
+
+
+class Path(sciline.Scope[FilenameType, str], str):
```

Comment:
`Path` will very commonly refer to `pathlib.Path`. Can we avoid using the same name (especially as we sometimes `from types import *`?)

Reply:
Renamed to `FilePath`.

> especially as we sometimes `from types import *`?

I want to get away from that. As you can see I have kept providers and types more structured here, no placing anything in a "meaningless" `types.py`. Still not there yet, but I hope it will work nice eventually.